### DISCUSSION TITLE
Dew heater module not working

### DISCUSSION THREAD

**Discussion by SkyDragon2023** [AllskyTeam/allsky-modules#121] [Q&A]
I installed the latest versions of your modules, i am running the current stable version of allsky, firstly I renamed the old folder as the git pull stated the old folder wasn't empty, after installing the modules I've noticed the dew heater module reports a false response in the module debug window and no data is shown on the overlay. I've restarted the pi as well. Any help is appreciated, the old adage of "if it ain't broken don't fix it" finally got me.

**Comment by SkyDragon2023** [comment]
Here is part of my log file:
Mar 24 09:45:40 allsky allsky[12537]: Traceback (most recent call last):
Mar 24 09:45:40 allsky allsky[12537]:   File "/home/pi/allsky/scripts/flow-runner.py", line 188, in <module>
Mar 24 09:45:40 allsky allsky[12537]:     _temp = importlib.import_module(moduleName)
Mar 24 09:45:40 allsky allsky[12537]:   File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
Mar 24 09:45:40 allsky allsky[12537]:     return _bootstrap._gcd_import(name[level:], package, level)
Mar 24 09:45:40 allsky allsky[12537]:   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
Mar 24 09:45:40 allsky allsky[12537]:   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
Mar 24 09:45:40 allsky allsky[12537]:   File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
Mar 24 09:45:40 allsky allsky[12537]: ModuleNotFoundError: No module named 'allsky_dewheater'

I uninstalled all the modules, renamed the folder and reinstated the old folder, went through the install stage, the dewheater module shows up in the module manager. the other modules seem to run ok, it's just this one that's causing issues.

**Reply by SkyDragon2023** [reply]
OK I found the log for the periodic jobs:
Mar 24 19:52:12 allsky allskperiodic[795]: INFO: Data acquired and written to extra data file openweather.json
Mar 24 19:52:12 allsky allskperiodic[795]: INFO: No last run info so assuming startup
Mar 24 19:52:12 allsky allskperiodic[795]: ERROR: Module allsky_dewheater.py failed on line 224 - Not running on a RPi!
Mar 24 19:52:12 allsky allskperiodic[795]: INFO: PI Status Data Written
Mar 24 19:53:27 allsky allskperiodic[1954]: INFO: Last run 75.59615731239319 seconds ago. Running every 900 seconds
Mar 24 19:53:27 allsky allskperiodic[1954]: ERROR: Module allsky_dewheater.py failed on line 224 - Not running on a RPi!
Mar 24 19:53:27 allsky allskperiodic[1954]: INFO: Will run in 44.54359173774719 seconds

**Comment by Alex-developer** [comment]
Can you please let me know the Pi version and OS version

Thanks

**Reply by SkyDragon2023** [reply]
It's a Pi 4b running bullseye.

**Reply by Alex-developer** [reply]
Sorry also meant to ask what version of Allsky are you running

Thanks

**Reply by SkyDragon2023** [reply]
It's the latest stable version on the home page

**Comment by Alex-developer** [comment]
Just tried this on a fresh install off bullseye on a pi4b and all is working,

Re reading your original post you mentioned renaming a folder, which folder did you rename as the module installation does not require anything to be renamed.

The error you are getting ' Not running on a RPi!' was fixed a while ago.

Thanks

Alex

**Reply by SkyDragon2023** [reply]
I had to rename the allky-modules folder that the git clone command creates as it stated the folder was not empty.
I'll do a fresh install of everything over the weekend, I wonder if something got messed up earlier during my build.

**Reply by Alex-developer** [reply]
Ah I see that make sense.

It does sound like there is a mismatch in some of the libraries, particularly the GPIO libraries leading to the module not being able to identify the board you are running.

Please let me know how the re install goes

Cheers

Alex
